RadixWallet/Clients/AccountsClient/AccountsClient+Live.swift

```diff
@@ -24,58 +24,94 @@ extension AccountsClient: DependencyKey {
+
+		let accountUpdates: AccountUpdates = { address in
+			await profileStore.accountValues().compactMap {
+				$0.first { $0.address == address }
+			}
+			.eraseToAnyAsyncSequence()
+		}
+
+		let newVirtualAccount: NewVirtualAccount = { request in
+			let networkID = request.networkID
+			let appearanceID = await nextAppearanceID(networkID, nil)
+			return try Profile.Network.Account(
+				networkID: networkID,
+				factorInstance: request.factorInstance,
+				displayName: request.name,
+				extraProperties: .init(appearanceID: appearanceID)
+			)
+		}
+
+		let getAccountByAddress: GetAccountByAddress = { address in
+			try await profileStore.network().entity(address: address)
+		}
+
+		let updateAccount: UpdateAccount = { updatedAccount in
+			try await profileStore.updating {
+				try $0.updateAccount(updatedAccount)
+			}
+		}
+
+		#if DEBUG
 		return Self(
 			getCurrentNetworkID: getCurrentNetworkID,
-			nextAccountIndex: nextAccountIndex,
+			nextAppearanceID: nextAppearanceID,
 			getAccountsOnCurrentNetwork: getAccountsOnCurrentNetwork,
-			getHiddenAccountsOnCurrentNetwork: {
-				try await profileStore.profile.network(id: getCurrentNetworkID()).getHiddenAccounts()
-			},
-			accountsOnCurrentNetwork: { await profileStore.accountValues() },
-			accountUpdates: { address in
-				await profileStore.accountValues().compactMap {
-					$0.first { $0.address == address }
-				}
-				.eraseToAnyAsyncSequence()
-			},
+			getHiddenAccountsOnCurrentNetwork: getHiddenAccountsOnCurrentNetwork,
+			accountsOnCurrentNetwork: accountsOnCurrentNetwork,
+			accountUpdates: accountUpdates,
 			getAccountsOnNetwork: getAccountsOnNetwork,
-			newVirtualAccount: { request in
-				let networkID = request.networkID
-				let numberOfExistingAccounts = await nextAccountIndex(networkID)
-				return try Profile.Network.Account(
-					networkID: networkID,
-					index: .init(numberOfExistingAccounts),
-					factorInstance: request.factorInstance,
-					displayName: request.name,
-					extraProperties: .init(numberOfAccountsOnNetwork: .init(numberOfExistingAccounts))
-				)
-			},
+			newVirtualAccount: newVirtualAccount,
 			saveVirtualAccounts: saveVirtualAccounts,
-			getAccountByAddress: { address in
-				try await profileStore.network().entity(address: address)
-			},
-			hasAccountOnNetwork: { networkID in
-				do {
-					let network = try await profileStore.profile.network(id: networkID)
-					// N.B. `accounts` is NonEmpty so `isEmpty` should always evaluate to `false`.
-					return network.hasSomeAccount()
-				} catch {
-					return false
-				}
-			},
-			updateAccount: { updatedAccount in
+			getAccountByAddress: getAccountByAddress,
+			hasAccountOnNetwork: hasAccountOnNetwork,
+			updateAccount: updateAccount,
+			debugOnlyDeleteAccount: { account in
 				try await profileStore.updating {
-					try $0.updateAccount(updatedAccount)
+					try $0.deleteAccount(account)
 				}
 			}
 		)
+		#else
```

Comment:
Pretty annoying to have to repeat everything, but I guess that's the only option

Reply:
Unless you make it a function or function variable and put if debug around it? It looks like it's not using anything defined in the init, so I guess it should work?

Reply:
the whole point is to not even define the function on AccountsClient outside of `DEBUG`. therefor, we must have two inits.

Reply:
Can't you put `#if DEBUG` around a `func`?

Reply:
resolve on slack, `profileStore` is not a stored property, so we cannot get away with anything better than this (if one does not cheat `ProfileStore.stored`)